### DISCUSSION TITLE
トラックボール用のコードの共通化

### DISCUSSION THREAD

**Discussion by koron** [Yowkees/keyball#13] [Ideas]
いまのトラックボール用のコードにはいくつかの問題があります。

* 同じようなコード、具体的には関数 `pointing_device_task()` のコピーが各keymapに分散してる
* `pointing_device_send()` を  `pointing_device_task()` で呼び出していないのでトラックボールが動かない
    * こういうのを直すのに全部の keymap.c を変更するのは手間

これらを解決するアイデアとして、共通のコードをrev1下に置いてしまうというアイデアがあります。
もし受け入れられそうなら、私の方で作業してPRしますがいかがでしょう?

**Comment by Yowkees** [comment]
実は今日私も同じく、アレ？最新のファームウェアでトラックボール動かない？と、うっすら気づいたのですが、あまりソフトに向き合う時間が作れず放置されてしまっていました。ご迷惑をおかけしました。

素晴らしいご提案ありがとうございます！
ご提案いただいたような大変更は、きっと僕には勇気が出せなかったと思います。大変お手数をおかけして申し訳ありませんが変更いただければとても嬉しいです！

**Reply by Yowkees** [reply]
マージ後じっくり勉強させていただきます！

**Reply by koron** [reply]
https://github.com/Yowkees/keyball/pull/14 にてPRいたしました。

ご検討ください。

**Comment by koron** [comment]
#15 で共通化がひと段落しました。
